pkg/controllers/inflightchecks/termination.go

```diff
@@ -37,9 +37,9 @@ func NewTermination(kubeClient client.Client) Check {
 	}
 }
 
-func (t *Termination) Check(ctx context.Context, node *v1.Node, provisioner *v1alpha5.Provisioner, pdbs *deprovisioning.PDBLimits) ([]Issue, error) {
+func (t *Termination) Check(ctx context.Context, node *v1.Node, machine *v1alpha5.Machine, provisioner *v1alpha5.Provisioner, pdbs *deprovisioning.PDBLimits) ([]Issue, error) {
```

Comment:
Here, the node is being looked up because it is used in every check. These inflight checks, at least currently, are validating against a combo of the Node and the Machine. Also, there is an event firing in the main controller. I think there might be room to improve this by not returning anything from the check and the just doing the event firing inside of the reconcile potentially in the future, but not something I'm necessarily thinking about covering in this PR